services/src/datasets/external/netcdfcf/schema/migration_0000_initial.sql

```diff
@@ -0,0 +1,248 @@
+    "time" "TimeInterval",
+    bbox "SpatialPartition2D",
+    resolution "SpatialResolution"
+);
+
+CREATE TYPE "GdalDatasetGeoTransform" AS (
+    origin_coordinate "Coordinate2D",
+    x_pixel_size double precision,
+    y_pixel_size double precision
+);
+
+CREATE TYPE "FileNotFoundHandling" AS ENUM (
+    'NoData',
+    'Error'
+);
+
+CREATE TYPE "RasterPropertiesKey" AS (
+    domain text,
+    key text
+);
+
+CREATE TYPE "RasterPropertiesEntryType" AS ENUM (
+    'Number',
+    'String'
+);
+
+CREATE TYPE "GdalMetadataMapping" AS (
+    source_key "RasterPropertiesKey",
+    target_key "RasterPropertiesKey",
+    target_type "RasterPropertiesEntryType"
+);
+
+CREATE DOMAIN "StringPair" AS text [2];
+
+CREATE TYPE "GdalRetryOptions" AS (
+    max_retries bigint
+);
+
+CREATE TYPE "GdalDatasetParameters" AS (
+    file_path text,
+    rasterband_channel bigint,
+    geo_transform "GdalDatasetGeoTransform",
+    width bigint,
+    height bigint,
+    file_not_found_handling "FileNotFoundHandling",
+    no_data_value double precision,
+    properties_mapping "GdalMetadataMapping" [],
+    gdal_open_options text [],
+    gdal_config_options "StringPair" [],
+    allow_alphaband_as_mask boolean,
+    retry "GdalRetryOptions"
+);
+
+CREATE TYPE "GdalLoadingInfoTemporalSlice" AS (
+    time "TimeInterval",
+    params "GdalDatasetParameters",
+    cache_ttl int
+);
+
+CREATE TYPE "GdalMetaDataList" AS (
+    result_descriptor "RasterResultDescriptor",
+    params "GdalLoadingInfoTemporalSlice" []
+);
+
+
+
+-- TABLE DEFINITIONS
+
+CREATE TABLE overviews (
+    file_name text PRIMARY KEY NOT NULL,
+    title text NOT NULL,
+    summary text NOT NULL,
+    spatial_reference "SpatialReference" NOT NULL,
+    colorizer "Colorizer" NOT NULL,
+    creator_name text,
+    creator_email text,
+    creator_institution text
+);
+
+CREATE TABLE groups (
+    file_name text NOT NULL REFERENCES overviews (
+        file_name
+    ) ON DELETE CASCADE DEFERRABLE,
+    name text [] NOT NULL,
+    title text NOT NULL,
+    description text NOT NULL,
+    data_type "RasterDataType",
+    data_range float [2],
+    unit text NOT NULL,
+
+    -- TODO: check if we need it
+    PRIMARY KEY (file_name, name) DEFERRABLE
+);
+
+CREATE TABLE entities (
+    file_name text NOT NULL REFERENCES overviews (
+        file_name
+    ) ON DELETE CASCADE DEFERRABLE,
+    id bigint NOT NULL,
+    name text NOT NULL,
+
+    -- TODO: check if we need it
+    PRIMARY KEY (file_name, id) DEFERRABLE
+);
+
+CREATE TABLE timestamps (
+    file_name text NOT NULL REFERENCES overviews (
+        file_name
+    ) ON DELETE CASCADE DEFERRABLE,
+    "time" bigint NOT NULL,
+
+    -- TODO: check if we need it
+    PRIMARY KEY (file_name, "time") DEFERRABLE
+);
+
+CREATE TABLE loading_infos (
+    file_name text NOT NULL REFERENCES overviews (
+        file_name
+    ) ON DELETE CASCADE DEFERRABLE,
+    group_names text [] NOT NULL,
+    entity_id bigint NOT NULL,
```

Comment:
shouldn't this reference entities?

Reply:
Do you mean I should add a compound foreign key from file_name and entity to entities?